### PR TITLE
GriptapeCloudVectorStoreDriver

### PR DESCRIPTION
**Pull request by cjkindel**

### CODE REVIEW

**Comment by collindutter** [comment]
@cjkindel is this ready for review? It's still marked as a draft.

**Comment by cjkindel** [comment]
> @cjkindel is this ready for review? It's still marked as a draft.

Ready for general review but not final form until 28 is finalized

File: griptape/drivers/vector/griptape_cloud_knowledge_base_vector_store_driver.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+        request: dict[str, Any] = {"query": query}
+        if count is not None:
+            request["count"] = count
+        if distance_metric is not None:
+            request["distance_metric"] = distance_metric
+        if filter is not None:
+            request["filter"] = filter
+        if include_vectors is not None:
+            request["include_vectors"] = include_vectors
```
Comment (collindutter) [inline griptape/drivers/vector/griptape_cloud_knowledge_base_vector_store_driver.py:100]: Maybe refactor to use [this syntax](https://github.com/griptape-ai/griptape/pull/951/files#diff-bf37af9210488ef8685c42ef0c8e19a8c6fb8829e00512171f1d45ebf1eb62edR136)?

File: tests/unit/drivers/vector/test_griptape_cloud_knowledge_base_vector_store_driver.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        },
+        {
+            "id": test_ids[1],
+            "vector": test_vecs[1],
+            "namespace": test_namespaces[1],
+            "meta": test_metas[1],
+            "score": test_scores[1],
+        },
+    ]
+}
```
Comment (collindutter) [inline tests/unit/drivers/vector/test_griptape_cloud_knowledge_base_vector_store_driver.py:37]: Move into `mock_requests_post`

File: tests/unit/drivers/vector/test_griptape_cloud_knowledge_base_vector_store_driver.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+
+class TestGriptapeCloudKnowledgeBaseVectorStoreDriver:
+    def mock_requests_post(*args, **kwargs):
+        return MockResponse(test_entries, 200)
```
Comment (collindutter) [inline tests/unit/drivers/vector/test_griptape_cloud_knowledge_base_vector_store_driver.py:42]: Mock how we do it [here](https://github.com/griptape-ai/griptape/blob/c9da271797dd69655f6a9b31de5be2da6ac3f827/tests/unit/tools/test_griptape_cloud_knowledge_base_client.py#L7-L23).

I assume you borrowed this style from [here](https://github.com/griptape-ai/griptape/blob/c9da271797dd69655f6a9b31de5be2da6ac3f827/tests/unit/tools/test_openweather_client.py#L29) which should probably be updated at some point.

**Review by collindutter — CHANGES_REQUESTED** [review]
Overall looks good! Minor test things, also please update docs.

File: docs/griptape-framework/drivers/vector-store-drivers.md [hunk 4]
```diff
@@ -49,6 +49,30 @@ print("\n\n".join(values))
 
 ```
 
+### Griptape Cloud Knowledge Base
+
+The [GriptapeCloudKnowledgeBaseVectorStoreDriver](../../reference/griptape/drivers/vector/griptape_cloud_knowledge_base_vector_store_driver.md) can be used to query data from a Griptape Cloud Knowledge Base. Loading into Knowledge Bases is not support, only querying. Here is a complete example of how the driver can be used to query an existing Knowledge Base:
```
Comment (collindutter) [inline docs/griptape-framework/drivers/vector-store-drivers.md:54]: `is not support` -> `is not supported at this time`.

`driver` -> `Driver`

File: tests/unit/drivers/vector/test_griptape_cloud_knowledge_base_vector_store_driver.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+class TestGriptapeCloudKnowledgeBaseVectorStoreDriver:
+    test_ids = [str(uuid.uuid4()), str(uuid.uuid4())]
+    test_vecs = [[0.1, 0.2, 0.3], [0.4, 0.5, 0.6]]
+    test_namespaces = [str(uuid.uuid4()), str(uuid.uuid4())]
+    test_metas = [{"key": "value1"}, {"key": "value2"}]
+    test_scores = [0.7, 0.8]
+
+    @pytest.fixture
+    def driver(self, mocker):
+        from griptape.drivers import GriptapeCloudKnowledgeBaseVectorStoreDriver
```
Comment (collindutter) [inline tests/unit/drivers/vector/test_griptape_cloud_knowledge_base_vector_store_driver.py:23]: Move to top level import

File: tests/unit/drivers/vector/test_griptape_cloud_knowledge_base_vector_store_driver.py [hunk 6]
```diff
@@ -0,0 +1,79 @@
+import pytest
+import uuid
+
+
+class MockResponse:
```
Comment (collindutter) [inline tests/unit/drivers/vector/test_griptape_cloud_knowledge_base_vector_store_driver.py:5]: Unused?

File: docs/griptape-framework/drivers/vector-store-drivers.md (hunk 4 above)
Comment (cjkindel) [reply docs/griptape-framework/drivers/vector-store-drivers.md:54]: Do we really want Driver capitalized, it isn't for rest of entries (can update all if we want)

**Review by collindutter — APPROVED** [review]
Thanks for the find and replace!

**Review by collindutter — APPROVED** [review]

**Review by collindutter — APPROVED** [review]

**Pull request merged**